### PR TITLE
Bug assignee for unmapped reources

### PR DESCRIPTION
**Pull request by v-soukum**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1 [hunk 1]
```diff
@@ -8,9 +8,14 @@ class BugMetaInfoProvider {
     hidden [bool] $BugLogUsingCSV = $false;
     hidden [string] $STMappingFilePath = $null
     hidden static $OrgMappingObj = @{}
+    hidden [PSObject] $emailRegEx
+    hidden static $AssigneeForUnmappedResources = @{}
 
     BugMetaInfoProvider() {
-
+        if ($null -eq $this.emailRegEx) {
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1:15]: every time will call LoadServerconfig.....

File: src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1 [hunk 2]
```diff
@@ -102,106 +107,317 @@ class BugMetaInfoProvider {
...
+                                $body.dataProviderContext.properties.permissionSetToken = "repoV2/$($ControlResult.ResourceContext.ResourceDetails.Project.id)/$($ControlResult.ResourceContext.ResourceDetails.id)"
+                                $accessList += $responseObj.dataProviders."ms.vss-admin-web.security-view-members-data-provider".identities | Where-Object { ($_.subjectKind -eq "user") -and (-not [string]::IsNullOrEmpty($_.mailAddress))} | ForEach-Object {
+                                    $identity = $_
+                                    $body.dataProviderContext.properties.subjectDescriptor = $_.descriptor
+                                    $identityPermissions = [WebRequestHelper]::InvokePostWebRequest($url, $body);
+                                    $configuredPermissions = @($identityPermissions.dataproviders."ms.vss-admin-web.security-view-permissions-data-provider".subjectPermissions | Where-Object {$_.permissionDisplayString -ne 'Not set'})
+                                    if ($configuredPermissions.Count -ge 12) {
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1:294]: logic ??

**Review by juhi037 — DISMISSED** [review]

File: src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1 [hunk 3]
```diff
@@ -102,106 +107,328 @@ class BugMetaInfoProvider {
...
+                            $responseObj = [WebRequestHelper]::InvokeGetWebRequest($apiURL);
+                            $roles =   @($responseObj | where {$_.role.displayname -eq 'Administrator'} |select identity) | where {(-not ($_.identity.displayname).Contains("\")) -and ($_.identity.displayname -notin @("GitHub", "Microsoft.VisualStudio.Services.TFS"))}
+                            if ($roles.Count -gt 0) {
+                                $userId = $roles[0].identity.id
+                                $assignee = $this.getUserFromUserId($organizationName, $userId)
+                            }
+                        }
+                        catch {
+                            $assignee = ""
+                            $this.PublishCustomMessage("Assignee Could not be determind.");
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1:153]: breaking here at PublishCustomMessage

File: src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1 [hunk 4]
```diff
@@ -102,106 +107,328 @@ class BugMetaInfoProvider {
...
+                {
+                    if ([Helpers]::CheckMember($ControlResult.ResourceContext.ResourceDetails, "lastModifiedBy")) {
+                        $assignee = $ControlResult.ResourceContext.ResourceDetails.lastModifiedBy.uniqueName
+                    }
+                }
+                
+                # if assignee is service account, then fetch assignee from jobs/permissions
+                if (($assignee -inotmatch [BugMetaInfoProvider]::emailRegEx.RegexList[0]) -or ([IdentityHelpers]::IsServiceAccount($assignee, 'User', [IdentityHelpers]::graphAccessToken))) 
+                {
+                    #$agentPoolsURL = "https://microsoftit.visualstudio.com/OneITVSO/_environments/385?view=resources&__rt=fps&__ver=2" -f $organizationName, $ControlResult.ResourceContext.ResourceGroupName, $ControlResult.ResourceContext.ResourceId.split('/')[-1]
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/BugLog/BugMetaInfoProvider.ps1:396]: remove commented api

File: src/AzSK.ADO/Framework/Configurations/SVT/ControlSettings.json [hunk 5]
```diff
@@ -773,6 +773,8 @@
   "BugLogging": {
     "BugLogAreaPath": "RootDefaultProject",
     "BugLogIterationPath": "RootDefaultProject",
+    "DefaultServiceId": "",
+    "GetAssigneeUsingFallbackMethod": true,
```
Comment (arvindsingh-cloud) [inline src/AzSK.ADO/Framework/Configurations/SVT/ControlSettings.json:777]: GetAssigneeUsingFallbackMethod should default false

**Review by arvindsingh-cloud — APPROVED** [review]

**Review by juhi037 — APPROVED** [review]

**Pull request merged**